### DISCUSSION TITLE
Transparent mode?

### DISCUSSION THREAD

**Discussion by D0wn10ad** [genotrance/px#205] [General]
Hi,

would like to know if a transparent mode could be considered?

i.e. all http(s) traffic will be re-routed to px even if it is not possible to configure proxy (there are quite some software that hardcode URLs into their programs - can't get away from those as they are g*vn issue)

 or it is already implemented? (I apologize if that is the case, I did go through the readme and px.ini trying to look for any hint of such feature....

**Comment by genotrance** [comment]
Px would need some minor changes to be used as a transparent HTTP proxy. OS or network firewall rules would also need to be configured to redirect all HTTP traffic to Px and that will need admin access.

HTTPS will be problematic though since the client will not be able to talk to or verify Px as the SSL server and code changes will be needed to make it work. It will also be a man-in-the-middle type setup which will compromise security.

Long story short - it's not on the roadmap and feels like something a commercial solution would be better suited for.
